### ISSUE TITLE
Provide a proxy server

### ISSUE THREAD

**Issue by jeff-zucker** [solid-contrib/pivot#12]
NSS provides a proxy server at https://solidcomunity.net/proxy?uri=SOME_URI.  This feature is needed to prevent blocking of request because of CORS issues.  Some examples of data that often can not be obtained from a client-side script without a proxy - RSS/Atom feeds, some ontologies, some specifications.

**Comment by thhck** [comment]
@jeff-zucker 
Could you point me where does this endpoint is implemented in NSS or tell me what needs to be done ?
Is it just fetching a URI given in parameter and rendering it directly ?

**Comment by bourgeoa** [comment]
There is no endpoint. It is a GET.
https://github.com/nodeSolidServer/node-solid-server/blob/5dc908c00ed63158e00393085258914d8e23e621/lib/create-app.js#L101

Follow the handler https://github.com/nodeSolidServer/node-solid-server/blob/5dc908c00ed63158e00393085258914d8e23e621/lib/create-app.js#L10

app.get
https://github.com/nodeSolidServer/node-solid-server/blob/5dc908c00ed63158e00393085258914d8e23e621/lib/handlers/cors-proxy.js#L65

**Comment by thhck** [comment]
Can you try:

https://github.com/thhck/css-proxy-module

If this is good enough we can add it to pivot.

I mapped the code from the previous NSS implementation. Please let me know if there is feature that need to be added. 

Also, if you have edge case for test that would be great.

**Comment by thhck** [comment]
Also I'm wondering what are the security consideration regarding such a proxy ?

**Comment by bourgeoa** [comment]
Just tested and at first looks great.

Why did you change the endpoint from `/proxy` to `/.proxy` ?
It was a parameter in NSS, but community.net used `/proxy`

@jeff-zucker 
you are the main user with your apps
Do you have a preference between `/proxy` and `/.proxy` ?
I shall add it on test on https://pivot-test.solidproject.org with your preference

**Comment by jeff-zucker** [comment]
> [@jeff-zucker](https://github.com/jeff-zucker) you are the main user with your apps Do you have a preference between `/proxy` and `/.proxy` ? I shall add it on test on https://pivot-test.solidproject.org with your preference

I'd prefer /proxy since that's what my code uses but changing that wouldn't be a major task so I could live with it.  We don't know if someone else might be using it so safer to stay with backwards compatibility.  Is it possible to look at the server logs to see where the hits to the endpoint come from?

**Comment by thhck** [comment]
> Why did you change the endpoint from /proxy to /.proxy ?

Because /proxy could be a pod called `proxy` . In CSS, all the server endpoint start with a dot I believe to avoide conflict ( `/.accoutns/` for example ) 

But this can be set in the config, so you just need to change this line:

https://github.com/thhck/css-proxy-module/blob/5f5649edbf08d4016b105971719aa82823224e81/config/proxy.json#L11

**Comment by thhck** [comment]
should I PR this feature ?

**Comment by bourgeoa** [comment]
@thhck 
Will it be complicated to only allow the proxy for authenticated users ?

The security issue was not really discussed but the ill-usage yes.
The conclusion was to allow use of proxy to resolve some cross-origin issue.
It is implemented in rdflib and display in SolidOS.

I have to check but I think it needs to be `/proxy` for SolidOS

**Comment by bourgeoa** [comment]
> should I PR this feature ?

not for the time being.
The best would be to have a way to change this parameter with an override,
and then may be there is a need for a PR.

We should also be able to avoid having `proxy` has a pod name. We need a black list

**Comment by thhck** [comment]
> The best would be to have a way to change this parameter with an override,
and then may be there is a need for a PR.

If by "this parameter" you mean the name of the endpoint , I think this is already the case you can set you our in the `config.json` file

> We should also be able to avoid having proxy has a pod name. We need a black list

As I said, is this an issue in solidcommunity.net if the pod name is in the subdomain ( a user creating a pod name `proxy` would create https://proxy.solidcommunity.net which wont conflict with the proxy url )

**Comment by bourgeoa** [comment]
> As I said, is this an issue in solidcommunity.net if the pod name is in the subdomain ( a user creating a pod name `proxy` would create https://proxy.solidcommunity.net which wont conflict with the proxy url )

Yes. But Pivot is able and will be used also in suffix situation and I also test pivot with suffix
